### PR TITLE
enable scheme in robots.txt host record

### PR DESCRIPTION
**Pull request by tyapkov**
Just created a new setting to enable HTTPS or HTTP on robots.txt HOST record. 
Will be great if somebody will review and if everything is ok merge it to the project

### CODE REVIEW

**Comment by yakky** [comment]
I can't find a formal spec for the host directive in which the scheme is required / recognized, could you provide a pointer?

**Comment by tyapkov** [comment]
I am not sure that it is required for all search enginges but I have followed the recommendations from Yandex (yandex.ru) in which they ask to add https:// to the host if site supports only secure connection. Here is the link https://yandex.ru/support/webmaster/controlling-robot/robots-txt.xml

**Comment by yakky** [comment]
@tyapkov ok. Please check the pep failure, then it's good to merge to me

**Comment by yakky** [comment]
@tyapkov you can rebase on master to fix some flake8 issues

**Comment by tyapkov** [comment]
I have refactored the code so now it passed Travis but shows some other checks failures. Can you please have a look at them?

**Comment by yakky** [comment]
@tyapkov they are actually coverage failure. Ideally we should add a test for the ``settings.USE_SCHEME_IN_HOST = True|False`` setting that check the corresponding value in the view output

**Comment by tyapkov** [comment]
@yakky ok. Who should make it?

**Comment by yakky** [comment]
@tyapkov for example you can add a check for ``Host`` like this 
https://github.com/jazzband/django-robots/blob/master/tests/test_views.py#L77
a second iteration overriding the default setting value and rerendering the view again (like in https://github.com/jazzband/django-robots/blob/master/tests/test_views.py#L74-L76)

something like:

```
self.assertTrue('Host: http://example.com' in content)

with self.settings(USE_SCHEME_IN_HOST=False):
    response = view_obj.render_to_response(context)
    response.render()
    content = force_text(response.content)
    self.assertTrue('Host: example.com' in content)
```

**Comment by tyapkov** [comment]
@yakky I still have code coverage problem. How can I solve it?

File: tests/test_views.py [hunk 1]
```diff
@@ -96,3 +96,33 @@ def test_view_site_2(self):
...
+        view_obj = RuleList()
+        view_obj.request = request
+        view_obj.current_site = Site.objects.get(pk=2)
+        view_obj.object_list = view_obj.get_queryset()
+        context = view_obj.get_context_data(object_list=view_obj.object_list)
+
+        with self.settings(USE_HOST=True):  
+            with self.settings(USE_SCHEME_IN_HOST=True):
+                response = view_obj.render_to_response(context)
+                esponse.render()
```
Comment (yakky) [inline tests/test_views.py]: ``r`` missing here and below

**Comment by yakky** [comment]
@tyapkov code looks good! check the missing ``r`` and then it should be all right! (you can test it locally by running ``tox`` command)

**Comment by tyapkov** [comment]
@yakky yes, I see it. The problem is that I edit with git online editor and my commits thread shows that it is somehow a challenge)

**Comment by tyapkov** [comment]
@yakky can you review the last commit one more time? I have added some lines

**Comment by yakky** [comment]
@tyapkov previous commit was ok. Due to the way settings are used in ``views.py`` the settings override does not work in the tests.
I'll refactor a bit the way settings are used and the code will work

**Comment by tyapkov** [comment]
@yakky ok, just leave a message if I need to change something

**Comment by yakky** [comment]
@tyapkov thanks for this. I fixed minor issues and merging this with #73  Closing this and merging your code in the other PR

**Pull request closed without merging**